### ISSUE TITLE
Error processing subtitles on ITV programme

### ISSUE THREAD

**Issue by jefro108** [Ascoware/get-iplayer-automator#155]
Downloading https://www.itv.com/hub/code-of-a-killer/2a3242a0001 Code of a Killer the subtitles aren't converted to srt.

```
Get iPlayer Automator 1.13 Initialized.
GetITVShows: ITV Cache Update Starting 
AppController: Updating Programme Index Feeds...

GetITVShows (Info): Episodes: 1336 Today Programmes: 174 
AppController: Index Updated.
GetITVShows: Update Finished


INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
AppController: Starting Downloads


Downloading Show 1/1:

ITV Download (ID=2a3242a0001): Downloading Code of a Killer - Tue Apr 03
INFO: Preparing Request for Auth Info
INFO: Requesting Metadata.
DEBUG: Metadata response status code: 200
INFO: Metadata processed.
DEBUG: youtube-dl args:["https://www.itv.com/hub/code-of-a-killer/2a3242a0001", "-f", "mp4", "--external-downloader-args", "-hide_banner", "-o", "/Volumes/media/Downloads/TV Shows/Code of a Killer/Code of a Killer.s01e01.Episode 1.mp4", "--write-sub"]
DEBUG: youtube-dl environment: ["PATH": "/Applications/Get iPlayer Automator.app/Contents/MacOS:/usr/bin"]
[ITV] 2a3242a0001: Downloading webpage

[ITV] 2a3242a0001: Downloading XML

[ITV] 2a3242a0001: Downloading JSON metadata

[ITV] 2a3242a0001: Downloading m3u8 information

ITV Download (ID=2a3242a0001): Subtitle path = /Volumes/media/Downloads/TV Shows/Code of a Killer/Code of a Killer.s01e01.Episode 1.en.ttml
[download] /Volumes/media/Downloads/TV Shows/Code of a Killer/Code of a Killer.s01e01.Episode 1.mp4 has already been downloaded


[download] 100% of 697.17MiB

ITV Download (ID=2a3242a0001): youtube-dl finished downloading
INFO: Downloading thumbnail
INFO: Thumbnail Download Completed
INFO: Beginning AtomicParsley Tagging.
INFO: AtomicParsley Tagging finished.
INFO: Converting Subtitles...
INFO: Converting to SubRip: /Volumes/media/Downloads/TV Shows/Code of a Killer/Code of a Killer.s01e01.Episode 1.en.ttml
ERROR: Conversion to SubRip failed: /Volumes/media/Downloads/TV Shows/Code of a Killer/Code of a Killer.s01e01.Episode 1.en.ttml
ffmpeg version 3.4.2-tessus Copyright (c) 2000-2018 the FFmpeg developers
  built with Apple LLVM version 9.0.0 (clang-900.0.39.2)
  configuration: --cc=/usr/bin/clang --prefix=/opt/ffmpeg --extra-version=tessus --enable-avisynth --enable-fontconfig --enable-gpl --enable-libass --enable-libbluray --enable-libfreetype --enable-libgsm --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopus --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-libzvbi --enable-version3 --pkg-config-flags=--static --disable-ffplay
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
/Volumes/media/Downloads/TV Shows/Code of a Killer/Code of a Killer.s01e01.Episode 1.en.ttml: Invalid data found when processing input

AppController: Downloads Finished

```

[Code of a Killer.s01e01.Episode 1.en.ttml.txt](https://github.com/Ascoware/get-iplayer-automator/files/1871660/Code.of.a.Killer.s01e01.Episode.1.en.ttml.txt)

**Comment by sleaux-meaux** [comment]
GiA doesn't instruct youtube-dl to convert the subtitles, so you only get the native format (TTML).

**Comment by hugofalkman** [comment]
@sleaux-meaux I used the downloaded srt file successfully as late as today .with the GiA download of Marcella s2e7.

**Comment by sleaux-meaux** [comment]
@hugofalkman: Marcella subtitles are WebVTT, not TTML, so can be converted by ffmpeg.

@jefro108: As a workaround, you can use the TTML-to-SRT converter bundled with GiA:

    "/Applications/Get iPlayer Automator.app/Contents/Resources/ttml2srt.py" file.ttml

**Comment by jefro108** [comment]
Thanks @sleaux-meaux - I just googled the issue and it seems it's something that's been sorted by youtube-dl already https://github.com/rg3/youtube-dl/issues/14191#issuecomment-329944436

**Comment by sleaux-meaux** [comment]
That is of no consequence since GiA doesn't use youtube-dl to convert the subtitles, at least not at present.

**Comment by skovatch** [comment]
It looked like ITV was using webvtt for all of its subtitles, so I just switched to ffmpeg for all subtitle conversions. But I see that's not the case, so I'll have to add a check for that and fallback to the old TTML to SRT converter.

**Comment by skovatch** [comment]
Fixed in v1.13.1.
